datadotworld/client.py

```diff
@@ -41,52 +47,344 @@ def __init__(self, token=None, propsfile="~/.data.world",
+
+        Returns
+        -------
+        SuccessMessage
+            Short message indicating success of the operation
+
+        Raises
+        ------
+        ApiException
+            If a server error occurs
+
+        Examples
+        --------
+        >>> dw.upload_files('jonloyens/an-intro-to-dataworld-dataset',
+        >>>                 ['/Users/jon/DataDotWorldBBall/DataDotWorldBBallTeam.csv'])
+        {'message': 'File(s) uploaded.'}
+        """
+        owner_id, dataset_id = dataset_key.split('/')
+        return self._uploads_api.upload_files(owner_id, dataset_id, files)
+
+    def delete_files(self, dataset_key=None, names=None):
+        """Delete dataset file(s)
+
+        Parameters
+        ----------
+        dataset_key : str
+            Dataset identifier, in the form of owner/id
+        files : list of str
+            The list of names for files to be deleted
+
+        Returns
+        -------
+        SuccessMessage
+            Short message indicating success of the operation
+
+        Raises
+        ------
+        ApiException
+            If a server error occurs
+
+        Examples
+        --------
+        >>> dw.delete_files(''jonloyens/an-intro-to-dataworld-dataset'', ['atx_startup_league_ranking.csv'])
+        {'message': 'Dataset file(s) have been successfully deleted.'}
+        """
+        owner_id, dataset_id = dataset_key.split('/')
+        return self._datasets_api.delete_files_and_sync_sources(owner_id, dataset_id, names)
+
+    def sync_files(self, dataset_key=None):
+        """Trigger synchronization process to update all dataset files linked to source URLs
+
+        Parameters
+        ----------
+        dataset_key : str
+            Dataset identifier, in the form of owner/id
+
+        Returns
+        -------
+        SuccessMessage
+            Short message indicating success of the operation
 
-    class Results:
-        def __init__(self, raw):
-            self.raw = raw
+        Raises
+        ------
+        ApiException
+            If a server error occurs
 
-        def __unicode__(self):
-            return self.as_string()
+        Examples
+        --------
+        >>> dw.sync_files('jonloyens/an-intro-to-dataworld-dataset')
+        {'message': 'Sync started.'}
+        """
+        owner_id, dataset_id = dataset_key.split('/')
+        return self._datasets_api.sync(owner_id, dataset_id)
 
-        def __repr__(self):
-            return "{0}\n...".format(self.as_string()[:250])
+    # Query Operations
 
-        def as_string(self):
-            return self.raw
+    def query(self, dataset_key, query, query_type="sql"):
+        """Query an existing dataset
 
-        def as_stream(self):
-            return StringIO(self.raw)
+        Parameters
+        ----------
+        dataset_key : str
+            Dataset identifier, in the form of owner/id
+        query : str
+            SQL or SPARQL query
+        query_type : {'sql', 'sparql'}, optional
```

Comment:
I'm just following this guide https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt in this case.